RE: Add C3 to linguist

rudy-in opened the pull request:
works on improving https://github.com/github-linguist/linguist/pull/7505
<!--- Briefly describe your changes in the field above. -->

## Description
This PR adds syntax highlighting and language detection support for the **C3 programming language**.

- Added a new language entry for C3 in `languages.yml`
- Added a grammar from the official [vscode-c3](https://github.com/c3lang/vscode-c3) repository
- Added samples for `.c3`, `.c3i`, `.c3t`
- Assigned a language color (`#2563eb`)

C3 is a modern systems programming language with syntax similar to C, but with improved safety and tooling.

---

## Checklist:

### I am adding a new language.

- [x] The extension of the new language is used in hundreds of repositories on GitHub.com.

  Search results for each extension:

  - `.c3`:  
    https://github.com/search?type=code&q=NOT+is%3Afork+path%3A*.c3+import

  - `.c3i`:  
    https://github.com/search?type=code&q=NOT+is%3Afork+path%3A*.c3i+import

  - `.c3t`:  
    https://github.com/search?type=code&q=NOT+is%3Afork+path%3A*.c3t+import

- [x] I have included a real-world usage sample for all extensions added in this PR:
  - Sample sources:
    - [Book-reader/fox33](https://github.com/Book-reader/fox33)
    - [joshring/argparse](https://github.com/joshring/argparse)
    - [lucy-language/lucy](https://github.com/lucy-language/lucy)
    - [NotsoanoNimus/uefi.c3l](https://github.com/NotsoanoNimus/uefi.c3l/) 
    - [m0tholith/c3terminal](https://github.com/m0tholith/c3terminal)
  - Sample licenses:
    - All samples are MIT licensed from the linked repositories.

- [x] I have included a syntax highlighting grammar:  
  https://github.com/c3lang/vscode-c3

- [x] I have added a color:
  - Hex value: `#2563eb`
  - Rationale: Chosen to distinguish C3 from C/C++ while retaining a similar cool tone.

---

### Notes

- All new sample files have explicit MIT licensing or are taken from MIT-licensed repositories.
- The grammar has been added as a submodule and points to the official upstream.
- Fixes https://github.com/github-linguist/linguist/issues/6072

cbuttner commented:
Hello Linguist maintainers, C3 contributor and author of the TM grammar here. I'm sorry this is a bit of a mess, looks like this is the third or fourth PR for this language. Some people open PR's without properly communicating with us and each other and sometimes end up abandoning things halfway.  Some don't know the language well enough yet and might get some things wrong. I actually wanted to do this PR myself but got busy and others beat me to it.
Anyway I appreciate your patience and I'm hoping we'll get this sorted out eventually.

@rudy-in You'll have to remove the `c3i` and `c3t` extensions including the example files, because if I understand correctly, they don't meet the usage requirements. Also you accidentally moved/deleted a file, you should fix that too.

cbuttner commented on vendor/licenses/git_submodule/vscode-c3.dep.yml:
```diff
@@ -1,18 +1,15 @@
 ---
-name: sublime-nginx
-version: f7a08a8c40caae3e423805dfe7b644d06c9ceff4
+name: vscode-c3
+version: 5202516e3f551a286fa8d7f940429b0fa8996fc4
```
There are new commits, please bump the version to the latest commit.

rudy-in commented:
I have removed .c3i and .c3t extensions along with their samples and updated the syntax highlighting for the language,
But i have a question to the linguist maintainers... , Why are submodules hard coded into linguist? i had to run many commands to ensure submodules were removed just to pull in the new changes from vscode-c3

had to run commands like
```
git rm -f vendor/grammars/vscode-c3
rm -rf .git/modules/vendor/grammars/vscode-c3
``` and then what not... it felt hacky but i had to do so much just to get in the changes or else when i used script/add-grammar it was reactivating the local git repo from the submodules which did not quite pull the changes made by @cbuttner

rudy-in commented:
in my opinion when updating the highlighting or something it shouldn't be so strict, what i expect is when i have to update i can simply remove the grammars dir and use `script/add-grammar` to update the grammar and it should freshly clone the repository again
